### PR TITLE
Template for Bitcoin Block 

### PR DESCRIPTION
**Pull request by lafinsd**
This Template overlays Bitcoin blocks as supported by the Bitcoin Core implementation which is the reference implementation. Bitcoin Core is available and currently supported on GitHub. There is no single spec I could find describing the format of the Bitcoin blocks. The format of the block has evolved as Bitcoin Improvement Proposals (BIPs) have been adopted. This template supports any implementation that uses the Bitcoin Core scheme for storing the blockchain.

### CODE REVIEW

File: HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme [hunk 1]
```diff
@@ -15,7 +15,7 @@
             <BuildableReference
                BuildableIdentifier = "primary"
                BlueprintIdentifier = "8D1107260486CEB800E47090"
-               BuildableName = "Hex Fiend.app"
+               BuildableName = "MYGH Hex Fiend.app"
```
Comment (kainjow) [inline HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme:18]: please revert

**Comment by kainjow** [comment]
@lafinsd is this still a work in progress? if so, you could put it in Draft mode until it's ready.

File: HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme (hunk 1 above)
Comment (lafinsd) [reply HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme]: Apologies. I am new to the GitHub collaborative environment. I'm trying to be a good citizen. Also, using GutHub Desktop I am seeing the same mess-up in Release.xcscheme as well. I think I have fixed both. GitHub Desktop is also noting changes in Debug.xcscheme.swp. I'm not sure how to proceed to do to fix that. Again, Apologies.

**Comment by lafinsd** [comment]
> @lafinsd is this still a work in progress? if so, you could put it in Draft mode until it's ready.

The Template BTCBlock.tcl is complete and usable now. 

In my Hex Fiend fork I have also checked in xBTCBlock.tcl. This an expanded version of BTCBlock.tcl but is a work in progress. It expands greatly on BTCBlock.tcl. It provides extensive breakdown and detail about specific objects in the bitcoin block whose interpretation depends on their content and context. My goal would be to replace BTCBlock.tcl with the expanded version eventually. 

I wrote the templates to help me learn about the technical details of the contents of the bitcoin block. BTCBlock.tcl is useful now so I wouldn't call it a draft. But the expanded one is problematic.  My concern is its accuracy. I am not a BTC expert so I cannot vouch for the accuracy of my labeling of the data objects and their contents. The only review mechanism I can see is to put it out there and let others who may know better collaborate and fix it or otherwise provide feedback. The expanded version is not ready for distribution unless there are Beta testers out there which I doubt. 

So I have one version that is ready now (BTCBlock.tcl, object of the Pull request) and one that is still a wip and may need to be vetted somehow unless it is issued as-is. I am not sure the best way to proceed and not sure how using the "draft" tag would make things better for everyone. I'll look up the documentation on draft mode.

**Comment by kainjow** [comment]
Ah I see, this pull request was created from the master branch of your fork. You could try creating a branch within your fork with just the BTCBlock.tcl file and create a PR here for that branch. I think that'd work?

File: HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme (hunk 1 above)
Comment (kainjow) [reply HexFiend_2.xcodeproj/xcshareddata/xcschemes/Debug.xcscheme]: I didn't realize at the time this PR was created from your master branch. See my comments below.

**Pull request closed without merging**